app/assets/javascripts/initializers/initializeDisplayAdVisibility.js

```diff
@@ -14,6 +14,110 @@ function initializeDisplayAdVisibility() {
       ad.classList.remove('hidden');
     }
   });
+}
+
+function trackAdImpression(adBox) {
+  var isBot = /bot|google|baidu|bing|msn|duckduckbot|teoma|slurp|yandex/i.test(
```

Comment:
Suggestion: could this regex be constantized somewhere? It's used twice in this PR, and I can see if being used much more!

Reply:
I agree that this could be better, however, this was extracted from [html variant impression tracking](https://github.com/forem/forem/blob/2b23d4f6ed0bd01119854a92b36ec2036282d0d3/app/assets/javascripts/initializers/initializeBaseTracking.js.erb#L130) and I think a more comprehensive refactor can happen in a later pass.